### ISSUE TITLE
Setup a Bountysource Salt campaign for funding

### ISSUE THREAD

**Issue by gnujedi** [ricochet-im/ricochet#537]
For this project to succeed in the plethora of messengers out there, funding is needed to keep developers working on this project.

[Bountysource](https://salt.bountysource.com/) created Salt for this purpose: 

> Salt is the sustained funding platform brought to you by Bountysource. Through Salt open-source projects can collect monthly contributions from supporters in order to reliably and predictably fund ongoing development.

Similar to Patreon, this allows pledges to FOSS developers. In today's age of surveillance, we need technologies like ricochet in order to maintain freedom. The Salt program will possibly help keep this project going.

**Comment by hsribei** [comment]
Good idea. [Open Collective](https://opencollective.com/) could be an alternative.

**Comment by Rspigler** [comment]
I completely agree, and would be willing to donate as indicated in this issue I posted today (https://github.com/ricochet-im/ricochet/issues/538) in order to encourage continued development.

While I agree the best case scenario is consistent funding and developing, and that current requested enhancements/feature requests such as file transfers (https://github.com/ricochet-im/ricochet/issues/15) and supporting synchronization of one account across multiple devices (https://github.com/ricochet-im/ricochet/issues/99) would be great improvements, they're not absolutely crucial.  For example, OnionShare (https://github.com/micahflee/onionshare) is a well respected application for file transfers, and while adding this functionality to Ricochet would be great, I believe keeping the code up to date is more important.

With a successful audit now complete (https://ricochet.im/files/ricochet-ncc-audit-2016-01.pdf), we know the code works and is secure.  The only real issue I feel is present in the code is the out of date Tor version, which currently would require consistent developer action to keep up to date without lag (Ricochet is currently using 0.2.8.9 which was released over 6 months ago).

In the short term I believe it would be best if we prepared for the worst case scenario, which would be low community donations and low continued enhancements.  If Tor were to be unbundled from Ricochet as discussed here (https://github.com/ricochet-im/ricochet/issues/538), it would allow users themselves to ensure they are running secure and up to date code by keeping their own installation of Tor up to date.

This short term fix would then allow us to focus on the long term goal - using Bountysource or Open Collective as discussed above to secure the long term development and success of Ricochet.

**Comment by Rspigler** [comment]
I can create the account for us if this is all is needed, but I believe someone better associated/trusted within this community should do so.

**Comment by gnujedi** [comment]
Also, Open Whisper Systems [BitHub](https://github.com/WhisperSystems/BitHub) could be an option. Sounds like a few of us at least would want to donate to this project to keep it going. I read that @s-rah has been recently [using it in her work](https://twitter.com/SarahJamieLewis/status/894002179526111233). It would be great to keep the work going on this project. @special, any thoughts on funding the project?

**Comment by increasingawareness** [comment]
So, I ~[set it up](https://salt.bountysource.com/teams/ricochet)~ (deleted, see below comment), somewhat on accident. I was seeing how far I could get with it and realized that I couldn't delete it. I signed a PGP message saying I'm not the owner. Not sure how to add people either. If people think we should continue with this, we can figure out how. Otherwise, I'll send an email to the bountysource salt team about the situation. Please weigh in on the situation, as I know money is a sensitive issue (I just want to be able to start funding this project so it keeps going).

Edit: Sent the Bounty Source team an email about the situation.

**Comment by Rspigler** [comment]
Hi @gnujedi, thanks for your work on this.  I don't believe I have enough standing to make a comment on ownership of the campaign, but I'm replying here to once again state my support for this project and my promise to donate.

**Comment by increasingawareness** [comment]
Okay, so I deactivated the team I had set up since there was already a "ricochet-im" team (not "ricochet") that has some bounties set up for this project, but the Salt feature of bountysource will have to wait until @special wants to set it up. I opened a pull request #554 to change the readme to include a bounty button to show the current amount of cash available to bounty hunters. This doesn't resolve this issue, as I intend lead devs to be able to be funded on a monthly basis. However, I hope it brings more people to work on the code.

Edit:
Here is the link to the bounty page: https://www.bountysource.com/teams/ricochet-im
